buildSrc/src/main/groovy/archunit.java-testing-conventions.gradle

```diff
@@ -12,6 +12,7 @@ dependencies {
     testImplementation dependency.junit5Jupiter
 
     testRuntimeOnly dependency.junit5VintageEngine
+    testRuntimeOnly dependency.junitPlatformLauncher
```

Comment:
Not sure I fully get this, why exactly does this break in `archunit-3rd-party-test`? AFAIS that module doesn't even have JUnit Jupiter tests, no? 🤔 
Not sure we should really add this globally, even if it solves that symptom, in particular if it's only that one specific test module.

Reply:
I've seen the
> `org.junit.platform.commons.JUnitException`: `OutputDirectoryProvider` not available; probably due to unaligned versions of the `junit-platform-engine` and `junit-platform-launcher` jars on the classpath/module path.

occur with the upgrade to JUnit 5.12 in other projects, too. I thought it was some kind of version conflict.  Do you know what Gradle's `useJUnitPlatform` actually provides? Some [Gradle upgrade notes](https://docs.gradle.org/current/userguide/upgrading_version_8.html#manually_declaring_dependencies) mention:

> If using JUnit 5, an explicit `runtimeOnly` dependency on `junit-platform-launcher` is required in addition to the existing `implementation` dependency on the test engine.

Reply:
Oh, and it turns out that not only `archunit-3rd-party-test` fails with this `JUnitException` – that just happened to be the first module where I saw the error. 🙄

Reply:
So I've dropped the misleading reference to `archunit-3rd-party-test` from the commit message.
If you prefer to understand the change with JUnit 5.12 better, we could also drop that update entirely from this PR, and have it in a separate one.

Reply:
Nah, if the docs say that you need that platform launcher dependency, then that's good enough for me 😉 Thanks a lot for the explanation!